src/main/java/core/basesyntax/impl/ReaderServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax.impl;
+
+import core.basesyntax.dao.OperationDaoImpl;
+import core.basesyntax.model.Operation;
+import core.basesyntax.model.Transaction;
+import core.basesyntax.service.ReaderService;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ReaderServiceImpl implements ReaderService {
+    private static final String FILE_NAME
+            = "src/main/java/core/basesyntax/resources/inputDataForPeriod.csv";
+    private static final String SEPARATE = ",";
+    private static final int INDEX_OPERATION = 0;
+    private static final int INDEX_FRUIT = 1;
+    private static final int INDEX_QUANTITY = 2;
+
+    private Transaction getFruitTransaction(String line) {
+        List<Operation> operations = new OperationDaoImpl().getListOperations();
```

Comment:
There are a couple of problems with this OperationDao and the task itself:
1. Your class is in fact not a DAO, you should not have one in this task, something like this should be more of a constant field and not a whole class with some interface.
2. you created some sort of a list, of `string operations` and their `arithmetic operations` and then use if to handle it, but it should be a proper strategy pattern implementation with: 
* `Operation` as an enum(represents our `balance`, `supply` etc.) with method `Operation forValue(String)`
* `OperationHandler` interface (has method `handle` or `apply` or `calculate`, which accepts our `Transaction` parameter) 
* 2 or more implementations of this hander(addition and subtraction) that will be processing the transactions in the right way and saving the results of that processing to the DB(which you don't have yet so let's add some mock like a `Storage` class with a Map of String fruit name and number of that fruits)
* and `OperationStrategy` interface with its implementation with a map filed of `Map<Operation, OperationHandler>` and method `OperationHandler get(Operation)` inside it

After all that you will have a clean and scalable code in other methods that are using this strategy